src/components/Graph/RainbowChart/RainbowChart.tsx

```diff
@@ -0,0 +1,185 @@
+import { Stack, Text } from '@chakra-ui/react'
+import { useColorModeValue } from '@chakra-ui/system'
+import { curveLinear } from '@visx/curve'
+import { ScaleSVG } from '@visx/responsive'
+import { AreaSeries, AreaStack, Axis, Tooltip, XYChart } from '@visx/xychart'
+import { extent, Numeric } from 'd3-array'
+import dayjs from 'dayjs'
+import omit from 'lodash/omit'
+import React, { useMemo } from 'react'
+import { useSelector } from 'react-redux'
+import { Amount } from 'components/Amount/Amount'
+import { AssetIcon } from 'components/AssetIcon'
+import { RainbowData } from 'hooks/useBalanceChartData/useBalanceChartData'
+import { selectAssets, selectSelectedLocale } from 'state/slices/selectors'
+import { useAppSelector } from 'state/store'
+import { colors } from 'theme/colors'
+
+export type RainbowChartProps = {
+  data: RainbowData[]
+  width: number
+  height: number
+  margin?: { top: number; right: number; bottom: number; left: number }
```

Comment:
`CSSProperties["margin"]` ?

Reply:
couldn't use `CSSProperties['margin']` specifically as `XYChart` wants a tighter `number`-only version of margin, but i could use their exported `Margin` type instead 👍